addons/mail/static/src/views/activity/activity_cell.js

```diff
@@ -0,0 +1,70 @@
+/** @odoo-module **/
+
+import { ActivityListPopover } from "@mail/activity/activity_list_popover";
+
+import { usePopover } from "@web/core/popover/popover_hook";
+import { useWowlService } from "@web/legacy/utils";
+
+import { Component, useRef, useSubEnv } from "@odoo/owl";
+
+export class ActivityCell extends Component {
+    setup() {
+        const messaging = useWowlService("mail.messaging");
+        useSubEnv(messaging.env); // force new env when used inside activity view which is still legacy
```

Comment:
Took me a bit of time to understand. The comment was not super clear, actually even mislead in the understanding.

So if I get it right, the whole chain from `ActivityRenderer` (included) -> `ActivityCell` (excluded) is legacy env.
These 2 LOCs allow having the same `env` as the one in `MessagingService`, which is basically `wowlEnv`

This problem doesn't exist in `ActivityListPopover` because it's being helped by `Popover` architecture that let it receive `wowlEnv`.